ssl/statem/extensions_srvr.c

```diff
@@ -866,9 +868,9 @@ int tls_parse_ctos_cookie(SSL_CONNECTION *s, PACKET *pkt, unsigned int context,
         SSLfatal(s, SSL_AD_INTERNAL_ERROR, ERR_R_INTERNAL_ERROR);
         return 0;
     }
-    if (!WPACKET_put_bytes_u8(&hrrpkt, SSL3_MT_SERVER_HELLO)
-            || !WPACKET_start_sub_packet_u24(&hrrpkt)
-            || !WPACKET_put_bytes_u16(&hrrpkt, TLS1_2_VERSION)
+    if (!ssl_set_handshake_header(s, &hrrpkt, SSL3_MT_SERVER_HELLO)
+            /* ServerHello msg start*/
+            || !WPACKET_put_bytes_u16(&hrrpkt, version1_2)
             || !WPACKET_memcpy(&hrrpkt, hrrrandom, SSL3_RANDOM_SIZE)
```

Comment:
@t8m @mattcaswell I'd appreciate your input on how to set the sequence number and fragment offset/length fields for dtls here.